Gems/ROS2/Code/Source/RobotImporter/Utils/RobotImporterUtils.cpp

```diff
@@ -0,0 +1,184 @@
+                }
+            }
+        };
+
+        const auto processLink = [&addFilenameFromGeometry, visual, colliders](const urdf::Link& link)
+        {
+            if (visual)
+            {
+                for (auto& p : link.visual_array)
+                {
+                    addFilenameFromGeometry(p->geometry);
+                }
+            }
+            if (colliders)
+            {
+                for (auto& p : link.collision_array)
+                {
+                    addFilenameFromGeometry(p->geometry);
+                }
+            }
+        };
+
+        std::function<void(const std::vector<urdf::LinkConstSharedPtr>&)> linkVisitor =
+            [&](const std::vector<urdf::LinkConstSharedPtr>& child_links) -> void
+        {
+            for (auto link : child_links)
+            {
+                processLink(*link);
+                std::vector<urdf::LinkConstSharedPtr> childVector(link->child_links.size());
+                std::transform(
+                    link->child_links.begin(),
+                    link->child_links.end(),
+                    childVector.begin(),
+                    [](const urdf::LinkSharedPtr& p)
+                    {
+                        return urdf::const_pointer_cast<urdf::Link>(p);
+                    });
+                linkVisitor(childVector);
+            }
+        };
+        linkVisitor({ rootLink });
+        return filenames;
+    }
+
+    /// Finds global path from URDF path
+    AZStd::string Utils::ResolveURDFPath(
+        AZStd::string unresolvedPath, const AZStd::string& urdfFilePath, const AZStd::function<bool(const AZStd::string&)>& fileExists)
+    {
+        if (unresolvedPath.starts_with("package://"))
+        {
+            AZ::StringFunc::Replace(unresolvedPath, "package://", "", true, true);
+            AZ::IO::Path urdfProperPath(urdfFilePath);
+            AZ::IO::Path package_path;
+            for (auto it = urdfProperPath.begin(); it != urdfProperPath.end(); it++)
+            {
+                package_path /= *it;
+                AZStd::string package_xml_candite = (package_path / "package.xml").String();
+                if (fileExists(package_xml_candite))
+                {
+                    // package.xml has been found
+                    return (package_path / unresolvedPath).String();
+                }
+            }
+            // we have nothing
+            return "";
+        }
+        if (unresolvedPath.starts_with("file:///"))
+        {
+            // seems to be global path
+            AZ::StringFunc::Replace(unresolvedPath, "file://", "", true, true);
+            return unresolvedPath;
+        }
+        // seems to be relative path
+        AZ::IO::Path relative_path(unresolvedPath);
+        AZ::IO::Path urdf_proper_Path(urdfFilePath);
+        AZ::IO::Path urdf_parent_path = urdf_proper_Path.ParentPath();
+        return (urdf_parent_path / relative_path).String();
```

Comment:
These variables should be lowerCamelCase.
```suggestion
        AZ::IO::Path relativePath(unresolvedPath)
        AZ::IO::Path urdfProperPath(urdfFilePath);
        AZ::IO::Path urdfParentPath = urdfProperPath.ParentPath();
        return (urdfParentPath / relativePath).String();